src/openloco/ui/scrollview.cpp

```diff
@@ -43,4 +45,174 @@ namespace openloco::ui::scrollview
+        WindowManager::invalidateWidget(w->type, w->number, widgetIndex);
+    }
+
+    // 0x004C8C7D
+    static void vTrackBottom(ui::window* const w, const int16_t scrollAreaIndex, const int8_t widgetIndex)
+    {
+        int16_t widgetHeight = w->widgets[widgetIndex].height() - 2;
+        if ((w->scroll_areas[scrollAreaIndex].flags & scroll_flags::HSCROLLBAR_VISIBLE) != 0)
+        {
+            widgetHeight -= SCROLLBAR_WIDTH + 1;
+        }
+        int16_t widgetContentHeight = std::max(w->scroll_areas[scrollAreaIndex].v_bottom - widgetHeight, 0);
+        w->scroll_areas[scrollAreaIndex].v_top = std::min<int16_t>(w->scroll_areas[scrollAreaIndex].v_top + widgetHeight, widgetContentHeight);
+        scrollview::update_thumbs(w, widgetIndex);
+        WindowManager::invalidateWidget(w->type, w->number, widgetIndex);
+    }
+
+    // 0x004C8689
+    void scrollLeft(const int16_t x, const int16_t y, ui::window* const w, ui::widget_t* const widget, const int8_t widgetIndex)
+    {
+        ui::scrollview::scroll_part scrollArea;
+        int16_t outX, outY;
+        int32_t scrollAreaOffset;
+
+        ui::scrollview::get_part(w, widget, x, y, &outX, &outY, &scrollArea, &scrollAreaOffset);
+
+        _currentScrollArea = (uint16_t)scrollArea;
+        _currentScrollOffset = scrollAreaOffset;
+        int16_t scrollAreaIndex = scrollAreaOffset / sizeof(ui::scroll_area_t);
+
+        // Not implemented for any window
+        // window->call_22()
+        switch (scrollArea)
+        {
+            case ui::scrollview::scroll_part::view:
+                w->call_scroll_mouse_down(outX, outY, scrollAreaIndex);
+                break;
+            case ui::scrollview::scroll_part::hscrollbar_button_left:
+                hButtonLeft(w, scrollAreaIndex, widgetIndex);
+                break;
+            case ui::scrollview::scroll_part::hscrollbar_button_right:
+                hButtonRight(w, scrollAreaIndex, widgetIndex);
+                break;
+            case ui::scrollview::scroll_part::hscrollbar_track_left:
+                hTrackLeft(w, scrollAreaIndex, widgetIndex);
+                break;
+            case ui::scrollview::scroll_part::hscrollbar_track_right:
+                hTrackRight(w, scrollAreaIndex, widgetIndex);
+                break;
+            case ui::scrollview::scroll_part::vscrollbar_button_top:
+                vButtonTop(w, scrollAreaIndex, widgetIndex);
+                break;
+            case ui::scrollview::scroll_part::vscrollbar_button_bottom:
+                vButtonBottom(w, scrollAreaIndex, widgetIndex);
+                break;
+            case ui::scrollview::scroll_part::vscrollbar_track_top:
+                vTrackTop(w, scrollAreaIndex, widgetIndex);
+                break;
+            case ui::scrollview::scroll_part::vscrollbar_track_bottom:
+                vTrackBottom(w, scrollAreaIndex, widgetIndex);
+                break;
+            default:
```

Comment:
Is there any enum member remaining?

Reply:
Thumb vertical/horizontal and none are remaining.